mapie/quantile_regression.py

```diff
@@ -0,0 +1,323 @@
+    # check_verbose,
+    fit_estimator,
+)
+from ._compatibility import np_quantile
+from .regression import MapieRegressor
+
+
+class MapieQuantileRegressor(MapieRegressor):
+    valid_methods_ = ["quantile"]
+    valid_agg_functions_ = [None, "median", "mean"]
+    fit_attributes = [
+        "list_estimators",
+        "list_y_preds_calib",
+        "conformity_scores_",
+        "y_calib"
+    ]
+
+    link_estimator_quantile = {
+        "GradientBoostingRegressor": ["loss", "alpha"],
+        "QuantileRegressor": ["quantile", "quantile"]
+    }
+
+    def __init__(
+        self,
+        estimator: RegressorMixin = None,
+        method: str = "quantile",
+        cv: Optional[Union[int, str, BaseCrossValidator]] = None,
+        n_jobs: Optional[int] = None,
+        agg_function: Optional[str] = "mean",
+        verbose: int = 0,
+        alpha: float = 0.2
+    ) -> None:
+        self.alpha = alpha
+        super().__init__(
+            estimator=estimator,
+            method=method,
+            cv=cv,
+            n_jobs=n_jobs,
+            agg_function=agg_function,
+            verbose=verbose,
+        )
+        if n_jobs is not None:
+            raise NotImplementedError
+
+    def _check_alpha(
+        self,
+        alpha: float,
+    ) -> ArrayLike:
+        if isinstance(alpha, float) is False:
+            raise ValueError(
+                "Invalid alpha. Allowed values are floats."
+            )
+        if ((alpha >= 0.5) or (alpha <= 0)):
+            raise ValueError(
+                "The alpha value has to be lower than 0.5."
+                "Recall that this is a two sided value."
+            )
+        # So we don't need these right?
+        alpha = [alpha/2, 1 - alpha/2]
+        alpha[0] = cast(Optional[NDArray], check_alpha(alpha[0]))[0]
+        alpha[1] = cast(Optional[NDArray], check_alpha(alpha[1]))[0]
+        return alpha
+
+    def _check_estimator(
+        self, estimator: Optional[RegressorMixin] = None
+    ) -> RegressorMixin:
+        if estimator is None:
+            return QuantileRegressor(solver="highs")
+        if not (hasattr(estimator, "fit") and hasattr(estimator, "predict")):
+            raise ValueError(
+                "Invalid estimator. "
+                "Please provide a regressor with fit and predict methods."
+            )
+        name_estimator = estimator.__class__.__name__
+        if name_estimator != "QuantileRegressor":
+            if name_estimator in self.link_estimator_quantile:
+                param_estimator = estimator.get_params()
+                if (
+                    self.link_estimator_quantile[name_estimator][0]
+                    in param_estimator
+                ):
+                    if param_estimator[
+                        self.link_estimator_quantile[
+                            name_estimator][0]] != "quantile":
+                        raise ValueError(
+                            "You need to set the loss/metric"
```

Comment:
To be more precise, "You need to set the loss/metric of your base model to `quantile`."

Reply:
done